vyper/builtin_functions/functions.py

```diff
@@ -309,6 +340,14 @@ def fetch_call_return(self, node):
 
         return return_type
 
+    def infer_arg_types(self, node):
+        validate_call_args(node, 3)
+
```

Comment:
maybe add an `assert len(node.args) == len(self._inputs)`, i always get a little nervous when i see zip without this guard.